File: app/src/main/java/com/lumiyaviewer/lumiya/react/RequestProcessor.java
package com.lumiyaviewer.lumiya.react;

import com.lumiyaviewer.lumiya.react.-$Lambda$s15PKVbd3BFZx563Ff4DOHT5V_w.AnonymousClass1;
import com.lumiyaviewer.lumiya.react.-$Lambda$s15PKVbd3BFZx563Ff4DOHT5V_w.AnonymousClass2;
import java.util.concurrent.Executor;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public abstract class RequestProcessor<K, Tup, Tdown> implements RequestHandler<K>, RequestSource<K, Tdown>, ResultHandler<K, Tdown>, Refreshable<K> {
    @Nullable
    private final Executor executor;
    @Nullable
    private RequestHandler<K> requestHandler;
    @Nonnull
    private final ResultHandler<K, Tup> resultHandler;

    public RequestProcessor(@Nonnull RequestSource<K, Tup> requestSource, @Nullable Executor executor) {
        this.executor = executor;
        this.resultHandler = requestSource.attachRequestHandler(this);
    }

    private void processRequestInternal(@Nonnull K k) {
        Object processRequest = processRequest(k);
        if (processRequest != null) {
            this.resultHandler.onResultData(k, processRequest);
        }
        if (!isRequestComplete(k, processRequest) && this.requestHandler != null) {
            this.requestHandler.onRequest(k);
        }
    }

    private void requestUpdateInternal(@Nonnull K k) {
        if (this.requestHandler != null) {
            this.requestHandler.onRequest(k);
        }
    }

    public ResultHandler<K, Tdown> attachRequestHandler(@Nonnull RequestHandler<K> requestHandler) {
        this.requestHandler = requestHandler;
        return this;
    }

    public void detachRequestHandler(@Nonnull RequestHandler<K> requestHandler) {
        if (this.requestHandler == requestHandler) {
            this.requestHandler = null;
        }
    }

    protected boolean isRequestComplete(@Nonnull K k, Tup tup) {
        return tup != null;
    }

    /* renamed from: lambda$-com_lumiyaviewer_lumiya_react_RequestProcessor_1507 */
    /* synthetic */ void m58lambda$-com_lumiyaviewer_lumiya_react_RequestProcessor_1507(Object obj) {
        requestUpdateInternal(obj);
    }

    /* renamed from: lambda$-com_lumiyaviewer_lumiya_react_RequestProcessor_2159 */
    /* synthetic */ void m59lambda$-com_lumiyaviewer_lumiya_react_RequestProcessor_2159(Object obj, Object obj2) {
        this.resultHandler.onResultData(obj, processResult(obj, obj2));
    }

    /* renamed from: lambda$-com_lumiyaviewer_lumiya_react_RequestProcessor_940 */
    /* synthetic */ void m60lambda$-com_lumiyaviewer_lumiya_react_RequestProcessor_940(Object obj) {
        processRequestInternal(obj);
    }

    public void onRequest(@Nonnull K k) {
        if (this.executor != null) {
            this.executor.execute(new -$Lambda$s15PKVbd3BFZx563Ff4DOHT5V_w(this, k));
        } else {
            processRequestInternal(k);
        }
    }

    public void onRequestCancelled(@Nonnull K k) {
        if (this.requestHandler != null) {
            this.requestHandler.onRequestCancelled(k);
        }
    }

    public void onResultData(@Nonnull K k, Tdown tdown) {
        if (this.executor != null) {
            this.executor.execute(new AnonymousClass2(this, k, tdown));
        } else {
            this.resultHandler.onResultData(k, processResult(k, tdown));
        }
    }

    public void onResultError(@Nonnull K k, Throwable th) {
        this.resultHandler.onResultError(k, th);
    }

    @Nullable
    protected abstract Tup processRequest(@Nonnull K k);

    protected abstract Tup processResult(@Nonnull K k, Tdown tdown);

    public void requestUpdate(K k) {
        if (this.executor != null) {
            this.executor.execute(new AnonymousClass1(this, k));
        } else {
            requestUpdateInternal(k);
        }
    }
}

File: app/src/main/kotlin/com/lumiyaviewer/lumiya/react/RequestProcessor.kt
package com.lumiyaviewer.lumiya.react

import java.util.concurrent.Executor

abstract class RequestProcessor<K, TUp, TDown>(
    requestSource: RequestSource<K, TUp>,
    private val executor: Executor?
) : RequestHandler<K>, RequestSource<K, TDown>, ResultHandler<K, TDown>, Refreshable<K> {

    private val resultHandler: ResultHandler<K, TUp> = requestSource.attachRequestHandler(this)
    private var requestHandler: RequestHandler<K>? = null

    protected abstract fun processRequest(key: K): TUp?
    
    protected abstract fun processResult(key: K, downstreamData: TDown): TUp

    protected open fun isRequestComplete(key: K, data: TUp?): Boolean = data != null

    private fun processRequestInternal(key: K) {
        val result = processRequest(key)
        if (result != null) {
            resultHandler.onResultData(key, result)
        }
        if (!isRequestComplete(key, result)) {
            requestHandler?.onRequest(key)
        }
    }

    private fun requestUpdateInternal(key: K) {
        requestHandler?.onRequest(key)
    }

    override fun attachRequestHandler(requestHandler: RequestHandler<K>): ResultHandler<K, TDown> {
        this.requestHandler = requestHandler
        return this
    }

    override fun detachRequestHandler(requestHandler: RequestHandler<K>) {
        if (this.requestHandler == requestHandler) {
            this.requestHandler = null
        }
    }

    override fun onRequest(key: K) {
        if (executor != null) {
            executor.execute {
                processRequestInternal(key)
            }
        } else {
            processRequestInternal(key)
        }
    }

    override fun onRequestCancelled(key: K) {
        requestHandler?.onRequestCancelled(key)
    }

    override fun onResultData(key: K, data: TDown) {
        if (executor != null) {
            executor.execute {
                resultHandler.onResultData(key, processResult(key, data))
            }
        } else {
            resultHandler.onResultData(key, processResult(key, data))
        }
    }

    override fun onResultError(key: K, error: Throwable) {
        resultHandler.onResultError(key, error)
    }

    override fun requestUpdate(key: K) {
        if (executor != null) {
            executor.execute {
                requestUpdateInternal(key)
            }
        } else {
            requestUpdateInternal(key)
        }
    }
}
